Content.Shared/Burial/BurialSystem.cs

```diff
@@ -0,0 +1,176 @@
+using Content.Shared.Burial;
+using Content.Shared.DoAfter;
+using Content.Shared.Interaction;
+using Content.Shared.Movement.Events;
+using Content.Shared.Placeable;
+using Content.Shared.Popups;
+using Content.Shared.Storage.Components;
+using Content.Shared.Storage.EntitySystems;
+using Content.Shared.Tools.Components;
+using Content.Shared.Tools.Systems;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Player;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Burial.Systems;
+
+public sealed class BurialSystem : EntitySystem
+{
+    [Dependency] private readonly SharedToolSystem _toolSystem = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly SharedEntityStorageSystem _storageSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<GraveComponent, InteractUsingEvent>(OnInteractUsing);
+        SubscribeLocalEvent<GraveComponent, ActivateInWorldEvent>(OnActivate);
+        SubscribeLocalEvent<GraveComponent, AfterInteractUsingEvent>(OnAfterInteractUsing, before: new[] { typeof(PlaceableSurfaceSystem) });
+        SubscribeLocalEvent<GraveComponent, GraveDiggingDoAfterEvent>(OnGraveDigging);
+
+        SubscribeLocalEvent<GraveComponent, StorageOpenAttemptEvent>(OnOpenAttempt);
+        SubscribeLocalEvent<GraveComponent, StorageCloseAttemptEvent>(OnCloseAttempt);
+        SubscribeLocalEvent<GraveComponent, StorageAfterOpenEvent>(OnAfterOpen);
+        SubscribeLocalEvent<GraveComponent, StorageAfterCloseEvent>(OnAfterClose);
+
+        SubscribeLocalEvent<GraveComponent, ContainerRelayMovementEntityEvent>(OnRelayMovement);
+        SubscribeLocalEvent<ActiveGraveComponent, ComponentShutdown>(OnStopDigging);
+    }
+
+    private void OnInteractUsing(EntityUid uid, GraveComponent component, InteractUsingEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        // allows someone to help you dig yourself out
+        if (TryComp<ActiveGraveComponent>(uid, out var active) && !active.DiggingSelfOut)
+            return;
+
+        if (TryComp<ToolComponent>(args.Used, out var shovel) && _toolSystem.HasQuality(args.Used, "Digging"))
+        {
```

Comment:
Use a component not tool qualities please.

Reply:
I saw that rolling pins were using tool qualities, so I had just copied that. I'll change this.